lib/fabrik/DirectorInstance.js

```diff
@@ -50,7 +50,10 @@ class DirectorInstance extends BaseInstance {
         this.networkSegmentIndex = networkSegmentIndex;
       })
       .tap(() => {
-        if (operation.type === 'delete') {
+        //TODO: This still has coupling with space guid as it calls ensureSpaceGuid. 
+        //As part of SF 2.0, we can go to ETCD and fetch the tenant_guid which will 
+        //ensure we dont call CF again. For now, we add a check for the feature validation.
+        if (operation.type === 'delete' && _.includes(this.manager.agent.features, 'backup')) {
```

Comment:
This check seems incorrect. We want to delete security groups for all instances irrespective of whether they support backup or not. May be this check should go elsewhere. (maybe within esnureSpaceGuid)

Reply:
Oh yes! Thanks. I am not adding it in ensureSpaceGuid, rather breaking promise.all and putting it in two chains.

Only problem is it will execute sequentially. put two codepath one having promise.all and one having only the deletesecgroup?

Dont want to do big change as with SF 2.0, ensurespaceguid would not be needed. Let me know what you suggest.